lib/grplatt.gi

```diff
@@ -2232,11 +2232,20 @@ InstallMethod(NormalSubgroups,"homomorphism principle perm groups",true,
 ##
 #M  Socle(<G>)
 ##
-InstallMethod(Socle,"from normal subgroups",true,[IsGroup],0,
+InstallMethod(Socle,"from normal subgroups",true,[IsGroup and IsFinite],0,
 function(G)
 local n,i,s;
   if Size(G)=1 then return G;fi;
-  # deal with lareg EA socle factor for fitting free
+
+  # force an IsNilpotent check
+  # should have and IsSolvable check, as well,
+  # but methods for solvable groups are only in CRISP
+  # which aggeressively checks for solvability, anyway
+  if IsNilpotentGroup(G) then
```

Comment:
This is assuming that the methods provided are static. One could imagine that the current method for nilpotent groups goes away at some point or for some strange reason becomes ranked lower. Then the problem would crop up. (It is not likely, but neither is the extra test hard.)